src/openMVG/sfm/pipelines/sfm_regions_provider.hpp

```diff
@@ -88,20 +88,25 @@ struct Regions_Provider
   virtual bool load(
     const SfM_Data & sfm_data,
     const std::string & feat_directory,
-    std::unique_ptr<features::Regions>& region_type)
+    std::unique_ptr<features::Regions>& region_type,
+    C_Progress& my_progress_bar = C_Progress::dummy())
   {
     region_type_.reset(region_type->EmptyClone());
 
-    C_Progress_display my_progress_bar( sfm_data.GetViews().size(),
-      std::cout, "\n- Regions Loading -\n");
+    my_progress_bar.restart(sfm_data.GetViews().size(), "\n- Regions Loading -\n");
     // Read for each view the corresponding regions and store them
-    bool bContinue = true;
+    volatile bool bContinue = true;
```

Comment:
`volatile` is no good for synchronization between threads on today's hardware that can do things like instruction reordering. See for example https://brooker.co.za/blog/2013/01/06/volatile.html or some threads on stack overflow discussing the differences.